MaterialComponent.js

```diff
@@ -53,23 +57,34 @@ export default class MaterialComponent extends Component {
     }
     return classText;
   }
+
   // Components must implement this method for their specific DOM structure
   materialDom(props) {
     return h("div", Object.assign({}, props), props.children);
   }
+
   render() {
     this.buildClassName();
     // Fetch a VNode
     const componentProps = this.props;
-    if (componentProps.class) {
-      // We delete class prop here so that any sub node's class doesn't get over-ridden from this
-      delete componentProps.class;
-    }
+    const userDefinedClasses =
+      componentProps.className || componentProps.class || "";
+    // We delete class props and add them later in the final
+    // step so every component does not need to handle user specified classes.
+    if (componentProps.class) delete componentProps.class;
```

Comment:
Might be better to not reference `class` as a dot notation as it's a [reserved word](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords). I highly doubt it would cause issues, but we sometimes have something called IE and it has its quirks.

```JavaScript
if (componentProps['class']) delete componentProps['class'];
```

Reply:
done